applications/DEMApplication/custom_processes/apply_forces_and_moments_process.cpp

```diff
@@ -118,11 +118,11 @@ namespace Kratos
 
         if(!mInterval.IsInInterval(time)) return;
 
-        block_for_each(mrModelPart.Nodes(), [&](Node<3>& rNode)
+        block_for_each(mrModelPart.Elements(), [&](Element& rElement)
         {
 
-            array_1d<double, 3>& force = rNode.FastGetSolutionStepValue(EXTERNAL_APPLIED_FORCE);
-            array_1d<double, 3>& moment = rNode.FastGetSolutionStepValue(EXTERNAL_APPLIED_MOMENT);
+            array_1d<double, 3>& force = rElement.GetGeometry()[0].FastGetSolutionStepValue(EXTERNAL_APPLIED_FORCE);
```

Comment:
this is not thread safe if you have more than one node
not sure if relevant

Reply:
In this case there's only one node per element, and it is not shared with other elements of any type.
A DEM particle has a central node, a rigid body wall has a central node.